electrumx/server/block_processor.py

```diff
@@ -434,20 +446,28 @@ def advance_txs(self, txs, is_unspendable):
                     continue
                 cache_value = spend_utxo(txin.prev_hash, txin.prev_idx)
                 undo_info_append(cache_value)
-                append_hashX(cache_value[:-15])
+                append_hashX(cache_value[:-TX_NUMB_LEN-TX_VALUE_LEN-IS_STAKE_FLAG_LEN])
 
             # Add the new UTXOs
+            staking_idx = -1
             for idx, txout in enumerate(tx.outputs):
                 # Ignore unspendable outputs
                 if is_unspendable(txout.pk_script):
+                    if len(txout.pk_script) == STAKING_HEADER_LENGHT:
```

Comment:
the whole script does not have a fixed length, due to the presence of `varint`, which can have a size of 1-9 bytes. The size of the whole script  can vary from 6 to 14 bytes

Reply:
fixed